File: code/C-sharp/TicTacToe.cs
using System;
using System.Collections.Generic;

namespace TicTacToe
{	
	class TicTacToe
	{
		public void generateAllGames(TicTacToeBoard board) {
			List<string> moves=board.legalMoves();
			foreach (string move in moves) {
				board.makeMove(move);
				if (board.isGameOver()) {
					board.playedMoves().ForEach(Console.Write);
					Console.WriteLine();
				} else {
					generateAllGames(new TicTacToeBoard(board));
				}
				board.undoLastMove();
			}
		}

	}

	class Program
	{
		static void Main(string[] args)
		{
			TicTacToe t=new TicTacToe();
			TicTacToeBoard b=new TicTacToeBoard();
			t.generateAllGames(b);
		}
	}
}

File: code/PHP/TicTacToe.php
<?php
require("TicTacToeBoard.php");

function generateAllGames($board) {
	global $count;
	foreach ($board->get_emptyPlaces() as $move) {
		$board->makeMove($move);
		if ($board->isGameOver() == true) {
			$count++;
			foreach ($board->get_playedMoves() as $move) {
				echo $move;
			}
			echo "\n";
			$board->toText();
		} else {
			generateAllGames($board);
		}
		$board->undoLastMove();
	}
}

$count=0;
$a_board = new TicTacToeBoard;
generateAllGames($a_board);
?>
